### PR TITLE
Ads

### PR DESCRIPTION
**Pull request by stlehmann**
**Description:**

Refers to closed PR https://github.com/home-assistant/home-assistant.github.io/pull/3777.

Add documentation for new ADS component.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):**
https://github.com/home-assistant/home-assistant/pull/10142

## Checklist:

  - [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`. 
  - [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Comment by frenck** [comment]
@MrLeeh Please specify the related parent PR in the start post.

File: source/_components/ads.markdown [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+The ADS (automation device specification) describes a device-independent and
+fieldbus-independent interface for communication between Beckhoff® automation
+devices running TwinCAT® and other devices implementing this interface.
+
+Configuration parameters:
+
+{% configuration %}
+
+- **device** (*Required*): The AMS NetId that identifies the device.
```
Comment (MartinHjelmare) [inline source/_components/ads.markdown]: This should be written using a kind of yaml syntax. See:
https://raw.githubusercontent.com/home-assistant/home-assistant.github.io/next/source/_components/dominos.markdown

File: source/_components/binary_sensor.ads.markdown [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+binary_sensor:
+  - platform: ads
+    adsvar: .boolean1
+```
+
+Configuration variables:
+
+{% configuration %}
+
+- **adsvar** (*Required*): The name of the variable which you want to access on
```
Comment (MartinHjelmare) [inline source/_components/binary_sensor.ads.markdown]: Syntax.

File: source/_components/light.ads.markdown [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  - platform: ads
+    adsvar: GVL.enable_light
+    adsvar_brightness: GVL.brightness
+```
+
+Configuration variables:
+
+{% configuration %}
+
+- **adsvar** (*Required*): The name of the boolean variable that switches the
```
Comment (MartinHjelmare) [inline source/_components/light.ads.markdown]: Syntax.

File: source/_components/sensor.ads.markdown [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    adsvar: GVL.temperature
+    unit_of_measurement: '°C'
+    adstype: int
+```
+
+Configuration variables:
+
+{% configuration %}
+
+- **adsvar** (*Required*): The name of the variable which you want to access on
```
Comment (MartinHjelmare) [inline source/_components/sensor.ads.markdown]: Syntax.

File: source/_components/switch.ads.markdown [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+# Example configuration.yaml entry
+switch:
+  - platform: ads
+    adsvar: .global_bool
+```
+
+Configuration variables:
+
+{% configuration %}
+- **adsvar** (*Required*): The name of the variable which you want to access on
```
Comment (MartinHjelmare) [inline source/_components/switch.ads.markdown]: Syntax.

File: source/_components/ads.markdown [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+title: "ADS"
+description: "Connect Home Assistant to TwinCAT devices via the ADS interface."
+date: 2016-05-25 23:39
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category: Hub
+ha_release: 0.59
```
Comment (MartinHjelmare) [inline source/_components/ads.markdown]: "0.60"

File: source/_components/binary_sensor.ads.markdown [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+title: "ADS Binary Sensor"
+description: "Instructions on how to set up ADS binary sensors within Home Assistant."
+date: 2017-10-25 10:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category: Binary Sensor
+ha_release: 0.59
```
Comment (MartinHjelmare) [inline source/_components/binary_sensor.ads.markdown]: "0.60"

File: source/_components/light.ads.markdown [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+title: "ADS Light"
+description: "Instruction on how to set up ADS lights within Home Assistant."
+date: 2017-10-25 10:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category: Light
+ha_release: 0.59
```
Comment (MartinHjelmare) [inline source/_components/light.ads.markdown]: "0.60"

File: source/_components/sensor.ads.markdown [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+title: "ADS Sensor"
+description: "Instructions how to integrate ADS numeric values into Home Assistant."
+date: 2017-10-25 10:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category: Sensor
+ha_release: 0.59
```
Comment (MartinHjelmare) [inline source/_components/sensor.ads.markdown]: "0.60"

File: source/_components/switch.ads.markdown [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+title: "ADS Switch"
+description: "Instructions on how to set up ADS switches within Home Assistant."
+date: 2017-10-25 10:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category: Switch
+ha_release: 0.59
```
Comment (MartinHjelmare) [inline source/_components/switch.ads.markdown]: "0.60"

File: source/_components/ads.markdown (hunk 6 above)
Comment (stlehmann) [reply source/_components/ads.markdown]: Done.

File: source/_components/ads.markdown (hunk 1 above)
Comment (stlehmann) [reply source/_components/ads.markdown]: Done.

File: source/_components/binary_sensor.ads.markdown (hunk 7 above)
Comment (stlehmann) [reply source/_components/binary_sensor.ads.markdown]: Done.

File: source/_components/binary_sensor.ads.markdown (hunk 2 above)
Comment (stlehmann) [reply source/_components/binary_sensor.ads.markdown]: Done.

File: source/_components/light.ads.markdown (hunk 8 above)
Comment (stlehmann) [reply source/_components/light.ads.markdown]: Done.

File: source/_components/light.ads.markdown (hunk 3 above)
Comment (stlehmann) [reply source/_components/light.ads.markdown]: Done.

File: source/_components/sensor.ads.markdown (hunk 4 above)
Comment (stlehmann) [reply source/_components/sensor.ads.markdown]: is there a *default* keyword for default values or should I just add the default value to the description?

File: source/_components/sensor.ads.markdown (hunk 9 above)
Comment (stlehmann) [reply source/_components/sensor.ads.markdown]: Done.

File: source/_components/switch.ads.markdown (hunk 10 above)
Comment (stlehmann) [reply source/_components/switch.ads.markdown]: Done.

File: source/_components/switch.ads.markdown (hunk 5 above)
Comment (stlehmann) [reply source/_components/switch.ads.markdown]: Done.

File: source/_components/sensor.ads.markdown (hunk 4 above)
Comment (MartinHjelmare) [reply source/_components/sensor.ads.markdown]: There is a `default` key, so use that.

File: source/_components/sensor.ads.markdown [hunk 11]
```diff
@@ -32,17 +32,24 @@ sensor:
...
+    type: string
+  adstype:
+    required: false
+    description: The datatype of the ADS variable, possible values are int, uint, byte
+    default: int
+    type: string
+  name:
+    required: false
+    description: An identifier for the sensor
+  factor:
```
Comment (MartinHjelmare) [inline source/_components/sensor.ads.markdown]: `name` (the key above `factor`) is missing the `type` key-value pair.

File: source/_components/sensor.ads.markdown [hunk 12]
```diff
@@ -32,17 +32,24 @@ sensor:
...
+    required: false
+    description: A factor that divides the stored value before displaying in Home Assistant
+    default: 1
 {% endconfiguration %}
```
Comment (MartinHjelmare) [inline source/_components/sensor.ads.markdown]: No `type` here for `factor` either.

File: source/_components/switch.ads.markdown [hunk 13]
```diff
@@ -26,10 +26,14 @@ switch:
...
-
 Configuration variables:
 
 {% configuration %}
-- **adsvar** (*Required*): The name of the variable which you want to access on
-the ADS device.
+  adsvar:
+    required: true
+    description: The name of the variable which you want to access on the ADS device
+  name:
```
Comment (MartinHjelmare) [inline source/_components/switch.ads.markdown]: Missing `type` for `adsvar`.

**Pull request merged**